Core/include/Acts/Seeding/SeedFinder.ipp

```diff
@@ -16,35 +16,36 @@ namespace Acts {
 
 template <typename external_spacepoint_t, typename platform_t>
 SeedFinder<external_spacepoint_t, platform_t>::SeedFinder(
-    Acts::SeedFinderConfig<external_spacepoint_t> config,
-    const Acts::SeedFinderOptions& options)
-    : m_config(config.toInternalUnits()), m_options(options.toInternalUnits()) {
+    const Acts::SeedFinderConfig<external_spacepoint_t>& config)
+    : m_config(config.toInternalUnits()) {
   // calculation of scattering using the highland formula
   // convert pT to p once theta angle is known
   m_config.highland = 13.6 * std::sqrt(m_config.radLengthPerSeed) *
                       (1 + 0.038 * std::log(m_config.radLengthPerSeed));
   float maxScatteringAngle = m_config.highland / m_config.minPt;
   m_config.maxScatteringAngle2 = maxScatteringAngle * maxScatteringAngle;
-
-  // helix radius in homogeneous magnetic field. Units are Kilotesla, MeV and
-  // millimeter
-  // TODO: change using ACTS units
-  m_config.pTPerHelixRadius = 300. * m_options.bFieldInZ;
-  m_config.minHelixDiameter2 =
-      std::pow(m_config.minPt * 2 / m_config.pTPerHelixRadius, 2);
-  m_config.pT2perRadius =
-      std::pow(m_config.highland / m_config.pTPerHelixRadius, 2);
-  m_config.sigmapT2perRadius =
-      m_config.pT2perRadius * std::pow(2 * m_config.sigmaScattering, 2);
 }
 
+template <typename external_spacepoint_t, typename platform_t>
+SeedFinder<external_spacepoint_t, platform_t>::SeedFinder() {}
+
 template <typename external_spacepoint_t, typename platform_t>
 template <template <typename...> typename container_t, typename sp_range_t>
 void SeedFinder<external_spacepoint_t, platform_t>::createSeedsForGroup(
-    SeedingState& state,
+    const Acts::SeedFinderOptions& options, SeedingState& state,
     std::back_insert_iterator<container_t<Seed<external_spacepoint_t>>> outIt,
     sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs,
     const Acts::Range1D<float>& rMiddleSPRange) const {
+  // helix radius in homogeneous magnetic field. Units are Kilotesla, MeV and
+  // millimeter
+  // TODO: change using ACTS units
+  const double pTPerHelixRadius = 300. * options.bFieldInZ;
+  const double minHelixDiameter2 =
+      std::pow(m_config.minPt * 2 / pTPerHelixRadius, 2);
+  const double pT2perRadius = std::pow(m_config.highland / pTPerHelixRadius, 2);
+  const double sigmapT2perRadius =
+      pT2perRadius * std::pow(2 * m_config.sigmaScattering, 2);
+
```

Comment:
Since createSeedsForGroup is called for every middle SP do you think we could perform some of these calculations only once?

Reply:
This is done. Please have a look.